openpype/hosts/houdini/plugins/publish/validate_sop_output_node.py

```diff
@@ -1,7 +1,14 @@
 # -*- coding: utf-8 -*-
 import pyblish.api
 from openpype.pipeline import PublishValidationError
+from openpype.pipeline.publish import RepairAction
 
+import hou
+
+
+class SelectInvalidAction(RepairAction):
+    label = "Select Invalid ROP"
+    icon = "mdi.cursor-default-click"
```

Comment:
Nicely picked icon! We might want to push that through all `SelectInvalidAction` actions if that's a preferred icon.

Anyway, I believe in this case you should be able to use the [action already existing for Houdini here](https://github.com/ynput/OpenPype/blob/b3044398fc9181db2d2230f9f0f5cc1de7e9d297/openpype/hosts/houdini/api/action.py#L7).

The logic you have now also wouldn't work for multiple instances since it'd only select one rop since the actions run over all instances as far as I know ;)

Reply:
> if multiple ROP nodes are selected, will artist figure out which ROP is associated with the selected validator error in publisher ?

It'll only select for invalid instances - **but** if there are two pointcache instances with the same invalidation then it'd select both. It's unfortunately how the Action system works, it triggers per plugin - not per instance.

> get_invalid return node paths instead of the node itself also it logs.

The logging is fine. Having it return the nodes instead of the paths seems fine to me if that's what SelectInvalidAction needs.

> I tried testing it first to learn how it works but for some reason it doesn't work on my machine, I suspect the process method is not even called!
> I added some log.debug inside process but no chance although get_invalid works fine in the mentioned validator above!

I think this is because `get_invalid` doesn't return a list of nodes but the path itself. It should [return a list of nodes to be correct inputs for `SelectInvalidAction`](https://github.com/ynput/OpenPype/blob/b3044398fc9181db2d2230f9f0f5cc1de7e9d297/openpype/hosts/houdini/api/action.py#L30-L35) if I look at this code. Looks like a bug in `validate_vdb_output_node.py`.

Reply:
> Having it return the nodes instead of the paths seems fine
so, there two things here. 
1. get invalid return the associated `output` node
2. my action selects the associated `rop` node 
what about having both ? 
- select invalid 🔍 
- select ROP 


> a bug in `validate_vdb_output_node.py`.
I'm going to fix it in another PR

Reply:
> select invalid 🔍
> select ROP

I guess it could be nice - but admittedly in that scenario I have no idea what "Select invalid" would select - haha. I'd just start off with solely implementing select invalid?

Reply:
> I'd just start off with solely implementing select invalid?

`select invalid 🔍` will select one of these 
- `hou.node(instance.data["instance_node"])` if output_node is None 
- `output_node` if not isinstance(output_node, hou.SopNode)
- `output_node` if node failed to cook `output_node.cook()` 
- `output_node` if not output_node.geometry() 

unlike `Select Invalid ROP`
will only selects the associated ROP for the selected instance

Reply:
Sure, let's give it a go.